scripts/lib/api/generateMdxComponents.ts

```diff
@@ -0,0 +1,361 @@
+    // Page is already dedicated to apiType; no heading needed
+    return props;
+  }
+  $(`<h3>${getLastPartFromFullIdentifier(id)}</h3>`).insertBefore($dl);
+
+  return {
+    ...props,
+    name: id.split(".").slice(-1)[0],
+  };
+}
+
+// ------------------------------------------------------------------
+// Generate MDX components
+// ------------------------------------------------------------------
+
+export async function generateMdxComponent(
+  apiType: ApiType,
+  props: componentProps,
+): Promise<string> {
+  const type = props.attributeType
+    ? props.attributeType.replaceAll("'", "&#x27;")
+    : undefined;
+  const value = props.attributeValue
+    ? props.attributeValue.replaceAll("'", "&#x27;")
+    : undefined;
+  const signature = await htmlSignatureToMd(props.signature!);
+  const extraSignatures: string[] = [];
+  for (const sig of props.extraSignatures ?? []) {
+    extraSignatures.push(`&#x27;${await htmlSignatureToMd(sig!)}&#x27;`);
+  }
+
+  return `<${apiType} 
+    id='${props.id}'
+    name='${props.name}'
+    attributeType='${type}'
+    attributeValue='${value}'
+    github='${props.githubSourceLink}'
+    signature='${signature}'
+    extraSignatures='[${extraSignatures.join(", ")}]'
+    >
+  `;
+}
+
+/**
+ * Removes the original link from sphinx.ext.viewcode and returns the HTML string for our own link.
+ *
+ * This returns the HTML string, rather than directly inserting into the HTML, because the insertion
+ * logic is most easily handled by the calling code.
+ *
+ * This function works the same regardless of whether the Sphinx build used `sphinx.ext.viewcode`
+ * or `sphinx.ext.linkcode` because they have the same HTML structure.
+ *
+ * If the link corresponds to a method, we only return a link if it has line numbers included,
+ * which implies that the link came from `sphinx.ext.linkcode` rather than `sphinx.ext.viewcode`.
+ * That's because the owning class will already have a link to the relevant file; it's
+ * noisy and not helpful to repeat the same link without line numbers for the individual methods.
+ */
+export function prepareGitHubLink(
+  $child: Cheerio<any>,
+  isMethod: boolean,
+): string {
+  const originalLink = $child.find(".viewcode-link").closest("a");
+  if (originalLink.length === 0) {
+    return "";
+  }
+  const href = originalLink.attr("href")!;
+  originalLink.first().remove();
+  return !isMethod || href.includes(".py#") ? href : "";
+}
+
+/**
+ * Find the element that both matches the `selector` and whose content is the same as `text`
+ */
+function findByText(
+  $: CheerioAPI,
+  $main: Cheerio<any>,
+  selector: string,
+  text: string,
+): Cheerio<any> {
+  return $main.find(selector).filter((i, el) => $(el).text().trim() === text);
+}
+
+export function mergeProps(
+  componentProps: componentProps,
+  props: componentProps[],
+): void {
+  componentProps.extraSignatures = [];
+  for (const prop of props) {
+    if (props.length == 0 || !prop.signature) {
```

Comment:
Why check `props.length == 0`? In that case, we won't enter the for loop.

Reply:
Oops :eyes: